### PR TITLE
Time Converter app

### PR DESCRIPTION
**Pull request by p2635**
Ask a colleague to review my code

### CODE REVIEW

File: 03-TimeConversionApp/TimeConverter/ContentView.swift [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+struct ContentView: View {
+  
+  @State private var inputValue = 60.0
+  @FocusState private var inputFocus
+  @State private var outputValue = 0.0
+  @State private var inputUnit = "Seconds"
+  @State private var outputUnit = "Minutes"
+  
+  private let options = ["Seconds", "Minutes", "Hours", "Days"]
```
Comment (jsanderson44) [inline 03-TimeConversionApp/TimeConverter/ContentView.swift]: I would look at using an `enum` for this. Much safer when switching over the options etc. 

You can either conform the enum to `String` this means the `.rawValue` will be a string representation of the case name. Or you could have a property on the enum to provide a title. e.g:

```
var title: String {
    switch self {
    case .seconds: return "Seconds"
    etc
    }
}
```

Comment (p2635) [reply 03-TimeConversionApp/TimeConverter/ContentView.swift]: **Phil**
Which option would you go for out of the two?

**John Sanderson**
Thinking about it, id maybe go for a mix of both

```swift
enum TimeUnit: String {
    case seconds
    case minutes
    etc

    var displayName: String {
        self.rawValue.capitalised
    }
}
```
That way the displayName property is a bit more descriptive about what it does and returns you a capitalised version of the string but still using the rawValue conformance means you dont have to explicitly type "Seconds" etc, which means if you were to add new cases you would just get the display name for "free"

**Phil:**
Never seen a var inside an enum before. So I use it like TimeUnit.seconds.displayname?

**John Sanderson:**
Yeah exactly. You probably dont even need the TimeUnit bit, so just .seconds.displayName.

Actually you wouldnt even need that, you can continue to use the ForEach and I think you can just do
```swift
ForEach(options, id: \.self) {
    Text($0.displayName)
}
```

If you conform the enum to CaseIterable as well (so enum TimeUnit: String, CaseIterable )
You can write your options property like so: `private let options = TimeUnit.allCases`

**Comment by p2635** [comment]
Applied the suggestions and successfully built/ran the app.

File: 03-TimeConversionApp/TimeConverter/TimeConverter/ContentView.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    
+    var displayName: String {
+      rawValue.capitalized
+    }
+  }
+  
+  private let options = TimeUnit.allCases
+  
+  // Convert to a base unit
+  private var getSeconds: Double {
```
Comment (jsanderson44) [inline 03-TimeConversionApp/TimeConverter/TimeConverter/ContentView.swift]: You could get even more swifty by adding another property to the enum. Say you have something like this in the enum: 
```
var secondsMultiplier: Double {
	switch self {
	case .seconds: return 1
	case .minutes: return 60
	case .hours: return 60 * 60
	case .days: return 60 * 60 * 24
	}
}
```

You could then remove this property completely and change the `convertedValue` property to: 
```
private var convertedValue: String {
	let converted = (inputValue * inputUnit.secondsMultiplier) / outputUnit.secondsMultiplier
	return converted.formatted()
}
```

The nice thing about this approach is that it means you only define the multipliers in one place

Comment (p2635) [reply 03-TimeConversionApp/TimeConverter/TimeConverter/ContentView.swift]: Implemented this incorrectly at first on with `outputUnit * outputUnit.secondsMultiplier`. It had a divide by zero infinity problem happening. Now it's all fixed on https://github.com/p2635/100-days-swift-ui/pull/9/commits/6bb9bf8315b8e4960900d37a9218facf049f96a6.

**Pull request merged**